--- CMakeLists.txt.orig	2018-04-06 03:48:37.000000000 -0500
+++ CMakeLists.txt	2018-04-15 14:34:46.000000000 -0500
@@ -43,6 +43,10 @@
 SetupDoxygen()
 if(NOT FREECAD_LIBPACK_USE OR FREECAD_LIBPACK_CHECKFILE_CLBUNDLER)
     SetupPython()
+    # This file is included directly due to some cMake macros that are defined within.
+    # If these macro definitions are embedded within another macro, it causes problems.
+    # MUST HAPPEN BEFORE PCL, BECAUSE THAT MAY INCLUDE QT VIA VTK!
+    include(cMake/FreeCAD_Helpers/SetupQt.cmake)
     SetupPCL()
     SetupPybind11()
     SetupXercesC()
@@ -64,9 +68,6 @@
     #find_package(OpenCV REQUIRED)
     SetupSwig()
     SetupEigen()
-    # This file is included directly due to some cMake macros that are defined within.
-    # If these macro definitions are embedded within another macro, it causes problems.
-    include(cMake/FreeCAD_Helpers/SetupQt.cmake)
     SetupFreetype()
 
     if(BUILD_GUI)
