--- src/App/FreeCADInit.py~	2022-12-06 19:35:37.000000000 -0700
+++ src/App/FreeCADInit.py	2024-01-11 01:04:11.000000000 -0700
@@ -108,6 +108,10 @@
     LibFcDir = os.path.realpath(LibFcDir)
     if (os.path.exists(LibFcDir) and not LibFcDir in libpaths):
         libpaths.append(LibFcDir)
+    if sys.platform == 'darwin':
+        BundleDir = os.path.realpath(FreeCAD.getHomePath()+'MacOS')
+        if os.path.exists(BundleDir):
+            libpaths.append(BundleDir)
     AddPath = FreeCAD.ConfigGet("AdditionalModulePaths").split(";")
     HomeMod = FreeCAD.getUserAppDataDir()+"Mod"
     HomeMod = os.path.realpath(HomeMod)
