--- cpack.d/CMakeLists.txt~	2019-12-25 11:44:39.000000000 -0700
+++ cpack.d/CMakeLists.txt	2023-06-04 15:17:03.000000000 -0600
@@ -119,7 +119,7 @@
   configure_file(DMGSetup.scpt.in DMGSetup.scpt @ONLY)
   configure_file("${CMAKE_SOURCE_DIR}/COPYING" "${CMAKE_BINARY_DIR}/COPYING.txt" COPYONLY)
   configure_file("${CMAKE_SOURCE_DIR}/README"  "${CMAKE_BINARY_DIR}/README.txt"  COPYONLY)
-  configure_file("${CMAKE_SOURCE_DIR}/packaging/macosx/Welcome.rtf"  "${CMAKE_BINARY_DIR}/WELCOME.rtf"  COPYONLY)
+  #configure_file("${CMAKE_SOURCE_DIR}/packaging/macosx/Welcome.rtf"  "${CMAKE_BINARY_DIR}/WELCOME.rtf"  COPYONLY)
 endif()
 
 foreach(FILE ${CONFIGS})
