--- modules/charset_macosxfs.c~	2015-02-22 07:11:32
+++ modules/charset_macosxfs.c	2024-10-31 18:23:15
@@ -202,7 +202,7 @@
  */
 static size_t macosxfs_encoding_pull(
 	void *cd,				/* Encoder handle */
-	char **inbuf, size_t *inbytesleft,	/* Script string */
+	const char **inbuf, size_t *inbytesleft,	/* Script string */
 	char **outbuf, size_t *outbytesleft)	/* UTF-16-LE string */
 {
 	static const int script_code = kCFStringEncodingUTF8;
@@ -324,7 +324,7 @@
 
 static size_t macosxfs_encoding_push(
 	void *cd,				/* Encoder handle */
-	char **inbuf, size_t *inbytesleft,	/* UTF-16-LE string */
+	const char **inbuf, size_t *inbytesleft,	/* UTF-16-LE string */
 	char **outbuf, size_t *outbytesleft)	/* Script string */
 {
 	static const int script_code = kCFStringEncodingUTF8;
