--- ocp/src/placer/PPlacement.h~	2008-01-07 09:22:51.000000000 -0800
+++ ocp/src/placer/PPlacement.h	2018-01-28 13:33:26.000000000 -0800
@@ -27,7 +27,12 @@
 #ifndef __PPLACEMENT_H
 #define __PPLACEMENT_H
 
-#ifdef __GNUC__
+#if __cplusplus > 201101L || defined __APPLE__
+#include <unordered_map>
+template<class K, class V, class H = std::hash<K>, class P = std::equal_to<K>,
+         class A = std::allocator<std::pair<const K, V> > >
+using hash_map = std::unordered_map<K, V, H, P, A>;
+#elif defined __GNUC__
 #if __GNUC__ < 3
 #include <hash_map.h>
 #else
--- ocp/src/placer/PPlacementFinal.cpp~	2011-02-06 06:37:33.000000000 -0800
+++ ocp/src/placer/PPlacementFinal.cpp	2018-01-28 14:00:15.000000000 -0800
@@ -24,18 +24,6 @@
 // Author : Christophe Alexandre  <Christophe.Alexandre@lip6.fr>
 //
 // Authors-Tag 
-#ifdef __GNUC__
-#if __GNUC__ < 3
-#include <hash_map.h>
-#else
-#include <ext/hash_map>
-#if __GNUC_MINOR__ == 0
-#else
-using namespace __gnu_cxx; // GCC 3.1 and later
-#endif
-#endif
-#endif
-
 #include <math.h>
 #include <unistd.h>
 #include <string.h>
