--- kdelibs-4.13.3/cmake/modules/FindKDE4Internal.cmake.orig	2014-11-16 18:37:57.000000000 -0700
+++ kdelibs-4.13.3/cmake/modules/FindKDE4Internal.cmake	2014-11-16 18:45:58.000000000 -0700
@@ -1116,7 +1116,7 @@
 endif (CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME STREQUAL GNU)
 
 if (UNIX)
-   set ( _KDE4_PLATFORM_DEFINITIONS "${_KDE4_PLATFORM_DEFINITIONS} -D_LARGEFILE64_SOURCE")
+   set ( _KDE4_PLATFORM_DEFINITIONS "${_KDE4_PLATFORM_DEFINITIONS} ")
 
    check_cxx_source_compiles("
 #include <sys/types.h>
@@ -1318,7 +1318,7 @@
 endif (CMAKE_COMPILER_IS_GNUCXX)
 
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
    _DETERMINE_GCC_SYSTEM_INCLUDE_DIRS(c++ _dirs)
    set(CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES
        ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES} ${_dirs})
@@ -1376,7 +1376,7 @@
        message("${_compile_output_var}")
        message(FATAL_ERROR "Qt compiled without support for -fvisibility=hidden. This will break plugins and linking of some applications. Please fix your Qt installation (try passing --reduce-exports to configure).")
    endif(NOT _compile_result)
-endif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+endif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
 
 if (CMAKE_C_COMPILER MATCHES "icc")
