--- meson.build~	2022-03-18 10:51:39.000000000 -0600
+++ meson.build	2023-01-28 16:21:50.000000000 -0700
@@ -293,14 +293,7 @@
 sse2_cflags = []
 if get_option('sse2')
   sse_prog = '''
-#if defined(__GNUC__)
-# if !defined(__amd64__) && !defined(__x86_64__)
-#   error "SSE2 intrinsics are only available on x86_64"
-# endif
-#elif defined (_MSC_VER) && !defined (_M_X64) && !defined (_M_AMD64)
-# error "SSE2 intrinsics not supported on x86 MSVC builds"
-#endif
-#if defined(__SSE__) || (_M_X64 > 0)
+#if defined(__SSE__) || (defined(_MSC_VER))
 # include <mmintrin.h>
 # include <xmmintrin.h>
 # include <emmintrin.h>
@@ -334,8 +327,6 @@
 #   error "GCC vector intrinsics are disabled on GCC prior to 4.9"
 # elif defined(__arm__)
 #   error "GCC vector intrinsics are disabled on ARM"
-# elif (__SIZEOF_POINTER__ < 8)
-#   error "GCC vector intrinsics are disabled on 32bit"
 # endif
 #else
 # error "Need GCC for GCC vectors intrinsics"
