--- ext/nokogiri/extconf.rb.orig	2021-09-15 22:20:13.000000000 +0300
+++ ext/nokogiri/extconf.rb	2021-09-15 22:21:00.000000000 +0300
@@ -614,7 +614,7 @@
 # append_cflags(["-Wcast-qual", "-Wwrite-strings"])
 
 # Add SDK-specific include path for macOS and brew versions before v2.2.12 (2020-04-08) [#1851, #1801]
-macos_mojave_sdk_include_path = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libxml2"
+macos_mojave_sdk_include_path = "@@PREFIX@@/include/libxml2"
 if config_system_libraries? && darwin? && Dir.exist?(macos_mojave_sdk_include_path)
   append_cppflags("-I#{macos_mojave_sdk_include_path}")
 end
@@ -1039,7 +1039,7 @@
 append_cppflags("-I#{File.join(libgumbo_recipe.path, "include")}")
 $libs = $libs + " " + File.join(libgumbo_recipe.path, "lib", "libgumbo.a")
 $LIBPATH = $LIBPATH | [File.join(libgumbo_recipe.path, "lib")]
-ensure_func("gumbo_parse_with_options", "nokogiri_gumbo.h")
+#ensure_func("gumbo_parse_with_options", "nokogiri_gumbo.h")
 
 have_func("xmlHasFeature") || abort("xmlHasFeature() is missing.") # introduced in libxml 2.6.21
 have_func("xmlFirstElementChild") # introduced in libxml 2.7.3
