--- akonadi/tests/CMakeLists.txt~	2014-11-04 13:19:27.000000000 -0800
+++ akonadi/tests/CMakeLists.txt	2017-08-06 12:20:09.000000000 -0700
@@ -100,12 +100,14 @@
 add_akonadi_test(collectionutilstest.cpp)
 add_akonadi_test(entitydisplayattributetest.cpp)
 add_akonadi_test(proxymodelstest.cpp)
-add_akonadi_test(actionstatemanagertest.cpp)
 add_akonadi_test(sharedvaluepooltest.cpp)
 add_akonadi_test(jobtest.cpp)
 add_akonadi_test(searchquerytest.cpp)
 
 # qtestlib tests that need non-exported stuff from akonadi-kde
+kde4_add_unit_test(actionstatemanagertest TESTNAME akonadi-actionstatemanagertest actionstatemanagertest.cpp ../pastehelper.cpp)
+target_link_libraries(actionstatemanagertest akonadi-kde akonadi-kmime akonaditestfake ${QT_QTTEST_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTGUI_LIBRARY} ${KDE4_KDECORE_LIBS} ${AKONADI_COMMON_LIBRARIES} ${QJSON_LIBRARIES})
+
 kde4_add_unit_test(resourceschedulertest TESTNAME akonadi-resourceschedulertest resourceschedulertest.cpp ../resourcescheduler.cpp)
 target_link_libraries(resourceschedulertest akonadi-kde ${QT_QTTEST_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${KDE4_KDECORE_LIBS} ${AKONADI_COMMON_LIBRARIES})
 
@@ -133,7 +135,7 @@
 add_akonadi_isolated_test(itemmovetest.cpp)
 add_akonadi_isolated_test(collectioncopytest.cpp)
 add_akonadi_isolated_test(collectionmovetest.cpp)
-add_akonadi_isolated_test(collectionsynctest.cpp)
+add_akonadi_isolated_test_advanced(collectionsynctest.cpp ../collectionsync.cpp "")
 add_akonadi_isolated_test(itemsynctest.cpp)
 add_akonadi_isolated_test(linktest.cpp)
 add_akonadi_isolated_test(cachetest.cpp)
--- akonadi/tests/actionstatemanagertest.cpp~	2014-11-04 13:19:27.000000000 -0800
+++ akonadi/tests/actionstatemanagertest.cpp	2017-08-06 13:09:05.000000000 -0700
@@ -27,7 +27,7 @@
 #include "../actionstatemanager.cpp"
 #undef QT_NO_CLIPBOARD
 
-#include "../pastehelper.cpp"
+#include "../pastehelper_p.h"
 
 typedef QHash<Akonadi::StandardActionManager::Type, bool> StateMap;
 Q_DECLARE_METATYPE( StateMap )
--- akonadi/tests/collectionsynctest.cpp~	2014-11-04 13:19:27.000000000 -0800
+++ akonadi/tests/collectionsynctest.cpp	2017-08-06 14:06:06.000000000 -0700
@@ -25,9 +25,10 @@
 #include <akonadi/collection.h>
 #include <akonadi/collectionfetchjob.h>
 #include <akonadi/collectionfetchscope.h>
+#include <akonadi/collectionmodifyjob.h>
 #include <akonadi/entitydisplayattribute.h>
 
-#include "../akonadi/collectionsync.cpp"
+#include "../akonadi/collectionsync_p.h"
 
 #include <krandom.h>
 
