--- buildtools/config.py.orig
+++ buildtools/config.py
@@ -235,10 +235,13 @@
             else:
                 for lst in [self.cflags, self.cxxflags]:
                     lst.append('-O3')
+            self.cflags.extend(os.environ.get('CFLAGS', '').split())
+            self.cxxflags.extend(os.environ.get('CXXFLAGS', '').split())
 
             lflags = self.getWxConfigValue('--libs')
             self.MONOLITHIC = (lflags.find("_xrc") == -1)
             self.lflags = lflags.split()
+            self.lflags.extend(os.environ.get('LDFLAGS', '').split())
 
             self.WXBASENAME = self.getWxConfigValue('--basename')
             self.WXRELEASE  = self.getWxConfigValue('--release')
@@ -840,7 +843,7 @@
                 line = line.strip()
                 endPos = line.rfind(' (')
                 curName = line[:endPos]
-                newName = '@loader_path/' + os.path.basename(curName)
+                newName = '@loader_path/../../../' + os.path.basename(curName)
                 cmd = 'install_name_tool -change %s %s %s' % (curName, newName, filename)
                 os.system(cmd)
 
@@ -1074,13 +1074,9 @@
         copy_file(opj(cfg.WXDIR, 'docs', filename), opj('license',filename),
                       update=1, verbose=1)
 
-    # Get the sip license too
-    copy_file(opj('sip', 'siplib', 'LICENSE'), opj('license', 'sip-license.txt'),
-              update=1, verbose=1)
-
     # Combine the relevant files into a single LICENSE.txt file
     text = ''
-    for filename in ['preamble.txt', 'licence.txt', 'lgpl.txt', 'sip-license.txt']:
+    for filename in ['preamble.txt', 'licence.txt', 'lgpl.txt']:
         with open(opj('license', filename), 'r', encoding='utf-8') as f:
             text += f.read() + '\n\n'
     with open('LICENSE.txt', 'w') as f:

