--- CMakeLists.txt~	2023-06-11 10:42:59.000000000 -0600
+++ CMakeLists.txt	2024-01-24 21:04:07.000000000 -0700
@@ -22,7 +22,6 @@
 endif()
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake")
-set(CMAKE_OSX_DEPLOYMENT_TARGET 10.13)
 
 # Use GNUInstallDirs to allow customisation
 # but set our own default data and bin dir
--- os/macosx/Info.plist.in~	2023-06-11 10:42:59.000000000 -0600
+++ os/macosx/Info.plist.in	2024-01-24 21:04:21.000000000 -0700
@@ -31,7 +31,5 @@
         <string>NSApplication</string>
         <key>NSHighResolutionCapable</key>
         <string>True</string>
-        <key>LSMinimumSystemVersion</key>
-        <string>10.13.0</string>
 </dict>
 </plist>
